aws/table_aws_rds_db_engine_version.go

```diff
@@ -0,0 +1,362 @@
+	// Create Session
+	svc, err := RDSClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_rds_db_engine_version.listRDSDBEngineVersions", "connection_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 20 {
+				maxLimit = 20
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &rds.DescribeDBEngineVersionsInput{
+		MaxRecords: aws.Int32(maxLimit),
+	}
+
+	if d.EqualsQualString("db_parameter_group_family") != "" {
+		input.DBParameterGroupFamily = aws.String(d.EqualsQualString("db_parameter_group_family"))
+	}
+	if d.EqualsQualString("engine") != "" {
+		input.Engine = aws.String(d.EqualsQualString("engine"))
+	}
+	if d.EqualsQualString("engine_version") != "" {
+		input.EngineVersion = aws.String(d.EqualsQualString("engine_version"))
+	}
+	if d.EqualsQuals["list_supported_character_sets"] != nil {
+		input.ListSupportedCharacterSets = aws.Bool(d.EqualsQuals["list_supported_character_sets"].GetBoolValue())
+	}
+	if d.EqualsQuals["list_supported_timezones"] != nil {
+		input.ListSupportedTimezones = aws.Bool(d.EqualsQuals["list_supported_timezones"].GetBoolValue())
+	}
+	if d.EqualsQuals["default_only"] != nil {
+		input.DefaultOnly = d.EqualsQuals["default_only"].GetBoolValue()
+	}
+
+	// Additional input filter
+	filters := buildEngineVersionInputFilter(d.EqualsQuals)
+	if len(filters) > 0 {
+		input.Filters = filters
+	}
+
+	paginator := rds.NewDescribeDBEngineVersionsPaginator(svc, input, func(o *rds.DescribeDBEngineVersionsPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		// apply rate limiting
+		d.WaitForListRateLimit(ctx)
+
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_rds_db_engine_version.listRDSDBEngineVersions", "api_error", err)
+			return nil, err
+		}
+
+		for _, item := range output.DBEngineVersions {
+			d.StreamListItem(ctx, item)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// TRANSFORM FUNCTIONS
+
+func getRDSDBEngineVersionTurbotTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	engineVersion := d.HydrateItem.(types.DBEngineVersion)
+
+	if engineVersion.TagList != nil {
+		turbotTagsMap := map[string]string{}
+		for _, i := range engineVersion.TagList {
+			turbotTagsMap[*i.Key] = *i.Value
+		}
+		return turbotTagsMap, nil
+	}
+	return nil, nil
+}
+
+func getRDSDBEngineVersionAka(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	engineVersion := d.HydrateItem.(types.DBEngineVersion)
+
+	if engineVersion.DBEngineVersionArn == nil {
```

Comment:
Can the `DBEngineVersionArn` value be null?

Reply:
Yes, @bigdatasourav, The APIs only provide the ARN value for the custom DB Engine Version not for the Engine Version which is AWS inbuilt.